armi/reactor/blueprints/assemblyBlueprint.py

```diff
@@ -174,15 +220,19 @@ def _createBlock(self, cs, blueprint, bDesign, axialIndex):
 
     def _checkParamConsistency(self, a):
         """Check that the number of block params specified is equal to the number of blocks specified."""
-        materialInputs = self.materialModifications or {}
         paramsToCheck = {
             "mesh points": self.axialMeshPoints,
             "heights": self.height,
             "xs types": self.xsTypes,
         }
-        for modName, modList in materialInputs.items():
-            paramName = "material modifications for {}".format(modName)
-            paramsToCheck[paramName] = modList
+
+        for mod in [self.materialModifications] + [
+            self.materialModifications.byComponent[key]
+            for key in self.materialModifications.byComponent.keys()
+        ]:
```

Comment:
Unless I'm missing something, I believe this might be more clearly written as:

```python
        for mod in [self.materialModifications] + list(self.materialModifications.byComponent.values()):
```

Reply:
Of course, will fix and put back up.